examples/rate-limiting/go.mod

```diff
@@ -0,0 +1,8 @@
+module github.com/modelcontextprotocol/go-sdk/examples/rate-limiting
```

Comment:
I guess we want a separate module because of x/time, right? Doc that in the commit message.
(I personally don't think it's such a big deal to include it, but if you feel it's best I'm okay with that.)

Reply:
Yes, documented in the commit message.